controller/token_blackbox_test.go

```diff
@@ -135,6 +169,37 @@ func (rest *TestTokenREST) checkServiceAccountCredentials(name string, id string
 	assert.True(rest.T(), token.IsSpecificServiceAccount(ctx, []string{name}))
 }
 
+func (rest *TestTokenREST) checkAuthorizationCode(name string, code string) {
+	service, _ := rest.SecuredController()
+	saToken, err := rest.dummyOauth.Exchange(service.Context, code)
+	assert.Nil(rest.T(), err)
+	assert.NotNil(rest.T(), saToken.TokenType)
+	assert.Equal(rest.T(), "bearer", saToken.TokenType)
+	assert.NotNil(rest.T(), saToken.AccessToken)
+}
+
+func (c *dummyOauth2Config) Exchange(ctx netcontext.Context, code string) (*oauth2.Token, error) {
+	claims := make(map[string]interface{})
+	accessToken, err := testtoken.GenerateTokenWithClaims(claims)
+	if err != nil {
+		panic(err)
+	}
+
+	var thirtyDays int64
+	thirtyDays = 60 * 60 * 24 * 30
+	token := &oauth2.Token{
+		TokenType:    "bearer",
+		AccessToken:  accessToken,
+		RefreshToken: "someRefreshToken",
```

Comment:
Generate some random strings (in suite or test setup) and use them as RefreshToken and AssessToken. Then validate these tokens in checkAuthorizationCode()